Migration guide 0.8-0.9 (#3604)
kyma-project/kyma: docs/migration-guides/0.8-0.9.md

> ## Migrate from v0.8 to v0.9

Reviewer: We agreed on the following attempt to migration guides: 
Include instructions for components that need to be handled manually by sys admins. If a component can be migrated automatically with no admin action required, do not write anything about it. 

Therefore in your case you should add an H2 heading with the component name to make it clear that this component needs admins attention. 

`Migrate from v0.8 to v0.9` should be the main heading of the document (H1) and under it the basic premise of this doc (if a component is not described - no action required, if described - follow the instructions) should be described.

Kyma 0.9.0 release ships a completely new version of the solution for handling content inside Kyma. The feature is called the Headless CMS and it introduces content management through custom resources. 

> > **TIP:** For more details, read release details in [GitHub](https://github.com/kyma-project/kyma/releases/tag/0.9.0) and on [Kyma Blog](https://kyma-project.io/blog/).

Reviewer 2: ```suggestion
> **TIP:** For more information, read the release details in [GitHub](https://github.com/kyma-project/kyma/releases/tag/0.9.0) and on [Kyma Blog](https://kyma-project.io/blog/).
```

> This change influences the Application Connector component, services specifications persistence. This change influences Kyma users that upgrade from 0.8.0 to 0.9.0 because the old location of the specifications is not supported by Application Connector.

Reviewer 3: "This change influences" repeated.

> To make migration as seamless as possible we introduce an automated migration script that does two things:

Reviewer 2: ```suggestion
To make the migration as seamless as possible, we introduced an automated migration script that does two things:
```

- removes old and registers new custom resources for the Headless CMS and the Asset Store components

> - migrates specifications registered with the Application Connector prior to 0.9.0 release into Headless CMS

Reviewer 2: ```suggestion
- migrates specifications registered with the Application Connector prior to 0.9.0 release into the Headless CMS
```

> **TIP:** For more details, read the migration [script](https://github.com/kyma-project/kyma/blob/master/resources/cms/templates/_helper_migrate_to_cms.txt).

### Migration script limitations

> The migration script does not address all possible use cases. Its scope focuses only on the use cases known as used by Kyma users.

Reviewer 2: ```suggestion
The migration script does not address all possible use cases. Its scope focuses only on those typical for Kyma users.
```

This means the script assumption is:
- One service has only one specification. There are no services with two different specs, like for example OpenAPI and AsyncAPI
- Documentation registration exposed through Metadata service is not used in current integrations.

The script basically takes specs from registered applications from location like `https://minio.{$DOMAIN}/content/service-class/{$SERVICE_CLASS_ID}/{$SPEC_NAME}.json` (where spec name is `asyncApiSpec` or `apiSpec`)
and saves them as `ClusterDocsTopic` custom resource.

### Manual migration

> Uncovered use cases require manual migration after a successful upgrade. Please familiarize with documentation for Headless CMS component to understand how to use it.

Reviewer 2: ```suggestion
Uncovered use cases require manual migration after a successful upgrade. Read the Headless CMS documentation to understand how to use it.
```
Can we add a link here?

> Keep in mind that prior 0.9.0 documentation was persisted in a single `content.json` file. Now, one of the options is to deliver content from `content.json` as separate files in a package.

Reviewer 2: ```suggestion
Keep in mind that prior to 0.9.0 release, documentation was kept in a single `content.json` file. Now, you can deliver content from separate `content.json` files in a package.
```

> In case of any issues with the migration please contact us on [Slack](http://slack.kyma-project.io) on `core-and-supporting` channel.

Reviewer 2: ```suggestion
In case of any issues with the migration, contact us on [Slack](http://slack.kyma-project.io) on the `core-and-supporting` channel.
```

### Recommendations 

> We recommend to not register documentation during application registration (property `documentation` under `/services` endpoint in Metadata service). We work now on a proposal to redesign the documentation registration payload (not specifications).

Reviewer 3: Please write more directly that the documentation registered by services endpoint won't show up in Service Catalog UI. It was hard for me to understand.

> We recommend to not register additional documentation, next too the API definitions, during application registration (property `documentation` under `/services` endpoint in Metadata service). We work now on a proposal to redesign the documentation registration payload (not specifications).

Reviewer 2: I think this section needs a little bit of rephrasing. What does `next to the API definitions` mean? And I don't like the brackets.

> We recommend to not register additional static documentation during application registration. This means we do not recommend to provide any data in `documentation` attribute in payload sent to `/services` endpoint from the Metadata service. We work now on a proposal to redesign the documentation registration payload (not specifications).

Reviewer 2: ```suggestion
We recommend not to register any additional static documentation during application registration. This means we do not recommend providing any data in the `documentation` attribute in a payload sent to the `/services` endpoint from the Metadata Service. We are working on a proposal to redesign the documentation registration payload (not specifications).
```

> At the moment if you use the API to register also static documentation we create a `ClusterDocsTopic` with a source of `docs` type that is not consumed the Console UI. You can use this temporary solution to understand how content registration is now implemented in Kyma.

Reviewer 2: - an API or the API? 
- This one is unclear for me:`we create a `ClusterDocsTopic` with a source of `docs` type`, what do you mean? 
- In what way can you use the solution to understand how content registration is now implemented in Kyma?

> For the time being, if you will use `documentation` attribute, and send content, the Metadata service creates a `ClusterDocsTopic` resource. Type of the source specified in the resource is set to `docs`. This type is not recognized by the Console UI. We implmeneted this temporary solution to not introduce any braking changes in current solution. Read the Headless CMS documentation to understand what `ClusterDocsTopic` resource is and what type of sources are supported.

Reviewer 2: ```suggestion
For the time being, if you use the `documentation` attribute and send content, the Metadata Service creates a `ClusterDocsTopic` resource. Type of the source specified in the resource is set to `docs`. This type is not recognized by the Console UI. We implemented this temporary solution to avoid introducing any breaking changes. Read the Headless CMS documentation to understand what `ClusterDocsTopic` resource is and what type of sources are supported.
```
